tensorflow/python/ops/gradients_util.py

```diff
@@ -386,25 +386,48 @@ def _SymGrad(op, out_grads):
 
 
 def _MaybeCompile(scope, op, func, grad_fn):
```

Comment:
I would rename this to `_run_with_attr_scope`. Or even better, refactor to have this function just return the attribute dictionary, and call it something like `get_extra_grad_attrs`. Then have the caller create the attr_scope and run `grad_fn`.

Reply:
Thanks for the suggestion; I refactored the function to just return the attribute dictionary, let me know if I should make any changes.